cdap-common/src/main/java/co/cask/cdap/common/internal/guava/ClassPath.java

```diff
@@ -201,22 +202,37 @@ public static ClassPath from(URI uri, ClassLoader classLoader) throws IOExceptio
    */
   @Beta
   public static class ResourceInfo {
+    private final URL baseURL;
     private final String resourceName;
     final ClassLoader loader;
 
-    static ResourceInfo of(String resourceName, ClassLoader loader) {
+    static ResourceInfo of(URL baseURL, String resourceName, ClassLoader loader) {
       if (resourceName.endsWith(CLASS_FILE_NAME_EXTENSION)) {
-        return new ClassInfo(resourceName, loader);
+        return new ClassInfo(baseURL, resourceName, loader);
       } else {
-        return new ResourceInfo(resourceName, loader);
+        return new ResourceInfo(baseURL, resourceName, loader);
       }
     }
   
-    ResourceInfo(String resourceName, ClassLoader loader) {
+    ResourceInfo(URL baseURL, String resourceName, ClassLoader loader) {
+      this.baseURL = baseURL;
       this.resourceName = checkNotNull(resourceName);
       this.loader = checkNotNull(loader);
     }
 
+    /**
+     * Returns the base URL of the resource.
+     * <p>
+     * For resource in local file, {@code url = baseURL + resourceName}.
+     * </p>
+     * <p>
+     * For resource in JAR file, {@code url = "jar:" + baseURL + "!/" + resourceName}
+     * </p>
+     */
+    public final URL baseURL() {
```

Comment:
actually just realized that this may be a guava class? If so, maybe we want to keep it similar. So ignore my comments...